### PR TITLE
Move location of gcloud gcs to storage

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description

File structure
- storage/
  - gcs/
    - object.go
    - object_test.go
    - request.go
    - bucket.go
    - errors.go
  - fake
     - bucket.go
     - bucket_test.go
     - testing
       - bucket_tests.go
       - register_bucket_tests.go
  - caching/
     - Moved everything from gcscaching to here.
  - storageutil/
    - Moved everything from gcsutil to here.
  - bucket_handle.go
  - bucket_handle_test.go
  - storage_handle.go
  - storage_handle_test.go
  - fake_storage_util.go
  - all other files
  - debug_bucket.go
  - mock_bucket.go

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - kokoro integration tests

### CODE REVIEW

File: internal/storage/fake/bucket.go [hunk 1]
```diff
@@ -26,7 +26,10 @@ import (
 	"strings"
 	"unicode/utf8"
 
-	"github.com/googlecloudplatform/gcsfuse/internal/gcloud/gcs"
+	"github.com/googlecloudplatform/gcsfuse/internal/storage"
+	bucket2 "github.com/googlecloudplatform/gcsfuse/internal/storage/bucket"
```
Comment (sethiay) [inline internal/storage/fake/bucket.go:30]: can we use bucket here instead of bucket2 ?

File: internal/storage/caching/integration_test.go [hunk 2]
```diff
@@ -12,16 +12,19 @@
...
 	"testing"
 	"time"
 
-	"github.com/googlecloudplatform/gcsfuse/internal/gcloud/gcs"
-	"github.com/googlecloudplatform/gcsfuse/internal/gcloud/gcs/gcscaching"
-	"github.com/googlecloudplatform/gcsfuse/internal/gcloud/gcs/gcsfake"
-	"github.com/googlecloudplatform/gcsfuse/internal/gcloud/gcs/gcsutil"
+	"github.com/googlecloudplatform/gcsfuse/internal/storage"
+	"github.com/googlecloudplatform/gcsfuse/internal/storage/bucket"
+	gcscaching2 "github.com/googlecloudplatform/gcsfuse/internal/storage/caching"
```
Comment (sethiay) [inline internal/storage/caching/integration_test.go:23]: can we use gcscaching here ?

Please check similar imports and correct them.

File: internal/storage/fake/bucket.go (hunk 1 above)
Comment (Tulsishah) [reply internal/storage/fake/bucket.go:30]: bucket name is used as variable name, that's why I kept bucket2

File: internal/storage/requests/requests.go [hunk 3]
```diff
@@ -12,13 +12,14 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package gcs
+package requests
```
Comment (vadlakondaswetha) [inline internal/storage/requests/requests.go:16]: keep the package name singular -> request

File: internal/storage/storagetesting/register_bucket_tests.go [hunk 4]
```diff
@@ -12,17 +12,17 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package gcstesting
+package storagetesting
```
Comment (vadlakondaswetha) [inline internal/storage/storagetesting/register_bucket_tests.go:15]: what are these tests and does these run?

Comment (Tulsishah) [reply internal/storage/storagetesting/register_bucket_tests.go:15]: This is for fake bucket unit tests. fake/bucket_test.go is calling these tests.

Comment (Tulsishah) [reply internal/storage/storagetesting/register_bucket_tests.go:15]: Should I move them under fake directory?

File: internal/storage/fake/testing/bucket_tests.go [hunk 5]
```diff
@@ -2870,22 +2870,22 @@ func (t *readTest) Ranges_EmptyObject() {
 	}
 
 	// Turn test cases into read requests.
-	var requests []*gcs.ReadObjectRequest
+	var reqs []*gcs.ReadObjectRequest
```
Comment (sethiay) [inline internal/storage/fake/testing/bucket_tests.go:2873]: keep requests here?

**Review by sethiay — APPROVED** [review]

**Pull request merged**